kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -251,9 +251,6 @@ public D edit() throws KubernetesClientException {
 
   @Override
   public R withName(String name) {
-    if (name == null || name.length() == 0) {
```

Comment:
we should skip this check only when generateName != null

Reply:
instead of name.length( ) == 0, use name.isEmpty( )